pincer/commands/interactable.py

```diff
@@ -0,0 +1,51 @@
+# Copyright Pincer 2021-Present
+# Full MIT License can be found in `LICENSE` at the project root.
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, TypeVar
+
+if TYPE_CHECKING:
+    from typing import Any, Awaitable, Callable
+
+T = TypeVar("T")
+
+
+class PartialInteractable(ABC):
```

Comment:
shouldn't be protocol?

Reply:
It implements `__call__` so I don't think so